apps/src/studio/studio.js

```diff
@@ -5511,7 +5511,7 @@ Studio.setMap = function(opts) {
   // Use the actual map for collisions, rendering, etc.
   Studio.wallMap = useMap;
   Studio.wallMapCollisions = true;
-  Studio.walls.setWallMapRequested(useMap);
+  Studio.walls.setWallMapRequested(mapValue);
```

Comment:
Is it safe to always use this value, given that the code above gets the chance to set a different `useMap` based on what `mapValue` is?  e.g. at https://github.com/code-dot-org/code-dot-org/pull/29989/files#diff-eeb7666ffd9b0becfc57bba206a98594R5490

Reply:
I think we want the value after "random" is converted to a specific string, but before "circle" is converted to "circle_nonjumbo". The `Studio.wallMap = useMap;` on line 5512 sets the map we use for collision.

Do you happen to remember the difference between `wallMapRequested` and `wallMap`?

Reply:
When we get to https://github.com/code-dot-org/code-dot-org/blob/227252020b364c029714bd5083613b83885a7319/apps/src/studio/studio.js#L5448, which is when we're setting a new background, it can happen that `wallMapRequested` is `"circle"` but `wallMap` is `"cicle_nonjumbo"`.  We pass in the former to `setMap` at that point and it freshly maps (no pun intended) this to the desired map (e.g. `"circle_nonjumbo"`) depending upon the new background.

Reply:
So the fix that you've made correctly caches the desired map name (as passed to `setMap`) for a future call to `setMap` when the background changes.